### PR TITLE
hsmd: Add fields to hsmd_sign_commitment_tx and hsmd_sign_remote_commitment_tx

### PR DESCRIPTION
**Pull request by ksedgwic**
Add to `hsmd_sign_remote_commitment_tx`:
- `commitment_number`
- `feerate`
- `htlcs`

Add to `hsmd_sign_commitment_tx`:
- `commitment_number`

These fields are needed for validating signers.

[Summary of all changes for validating signers](https://gitlab.com/lightning-signer/docs/-/wikis/C-Lightning-Modifications)

### CODE REVIEW

File: hsmd/Makefile [hunk 1]
```diff
@@ -31,11 +33,13 @@ HSMD_COMMON_OBJS :=				\
 	common/status_wiregen.o			\
 	common/hash_u5.o			\
 	common/hsm_encryption.o			\
+	common/htlc_wire.o			\
 	common/key_derive.o			\
 	common/lease_rates.o			\
 	common/memleak.o			\
 	common/msg_queue.o			\
 	common/node_id.o			\
+	common/onionreply.o			\
```
Comment (devrandom) [inline hsmd/Makefile:42]: stray?

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -176,6 +176,57 @@ u8 *p2wpkh_for_keyidx(const tal_t *ctx, struct lightningd *ld, u64 keyidx)
...
+				    channel->our_config.dust_limit, LOCAL,
+				    channel_has(channel, OPT_ANCHOR_OUTPUTS)))
+			continue;
+		struct simple_htlc *simple =
+		    new_simple_htlc(NULL,
+				    htlc_state_owner(hin->hstate),
+				    hin->msat,
+				    &hin->payment_hash,
+				    hin->cltv_expiry
+				    );
```
Comment (devrandom) [inline lightningd/peer_control.c:202]: I think code convention for this repo has this on the same line as the last arg

File: lightningd/test/Makefile [hunk 3]
```diff
@@ -13,12 +13,14 @@ LIGHTNINGD_TEST_COMMON_OBJS :=			\
...
 	common/daemon_conn.o			\
 	common/htlc_state.o			\
+	common/htlc_wire.o			\
 	common/json.o				\
 	common/key_derive.o			\
 	common/pseudorand.o			\
 	common/random_select.o			\
 	common/memleak.o			\
 	common/msg_queue.o			\
+	common/onionreply.o			\
```
Comment (devrandom) [inline lightningd/test/Makefile:23]: stray?

**Review by devrandom — APPROVED** [review]

**Comment by devrandom** [comment]
LGTM other than the above

File: hsmd/Makefile (hunk 1 above)
Comment (ksedgwic) [reply hsmd/Makefile:42]: Unfortunately not stray:
```
/usr/bin/ld: common/htlc_wire.o: in function `failed_htlc_dup':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:23: undefined reference to `dup_onionreply'
/usr/bin/ld: common/htlc_wire.o: in function `towire_failed_htlc':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:150: undefined reference to `towire_onionreply'
/usr/bin/ld: common/htlc_wire.o: in function `fromwire_failed_htlc':
/home/ksedgwic/lightning-signer/c-lightning/common/htlc_wire.c:265: undefined reference to `fromwire_onionreply'
```

If we broke `simple_htlc` out onto it's own file we could fix it, I think.  Worth doing?

File: lightningd/peer_control.c (hunk 2 above)
Comment (ksedgwic) [reply lightningd/peer_control.c]: fixed

File: lightningd/test/Makefile (hunk 3 above)
Comment (ksedgwic) [reply lightningd/test/Makefile:23]: same as above

File: hsmd/Makefile (hunk 1 above)
Comment (devrandom) [reply hsmd/Makefile:42]: ah, I see.  probably not worth it. C-L people can raise this if they think it's useful.

File: lightningd/peer_control.c [hunk 4]
```diff
@@ -176,6 +176,55 @@ u8 *p2wpkh_for_keyidx(const tal_t *ctx, struct lightningd *ld, u64 keyidx)
...
+	}
+	const struct htlc_out *hout;
+	struct htlc_out_map_iter outi;
+	for (hout = htlc_out_map_first(htlcs_out, &outi);
+	     hout;
+	     hout = htlc_out_map_next(htlcs_out, &outi)) {
+		if (hout->key.channel != channel)
+			continue;
+		if (hout->hstate < RCVD_ADD_REVOCATION || hout->hstate > RCVD_REMOVE_COMMIT)
+			continue;
```
Comment (ksedgwic) [inline lightningd/peer_control.c:212]: I'm not sure why we don't need the equivalent filter on `htlcs_in` above ...

Comment (devrandom) [reply lightningd/peer_control.c:212]: did you try to add the filter (with `s/RCVD/SENT/`)?  does it fail?

Comment (ksedgwic) [reply lightningd/peer_control.c]: We removed this code w/ the `sign_holder_commitment_tx` API cleanup

**Comment by devrandom** [comment]
ACK

**Comment by ksedgwic** [comment]
This PR is subsumed by https://github.com/ElementsProject/lightning/pull/4981 which is merged.

**Pull request closed without merging**